### PR TITLE
feat: add facebook support

### LINKED ISSUE #54: Add Facebook support
Facebook URLs would be useful for downloading content without an account or having to open an app to view.

Example URL formats:
https://fb.watch/gsa6WvY0sF/
https://www.facebook.com/buzzfeedapop/videos/1061711167873198/?extid=NS-UNK-UNK-UNK-AN_GK0T-GK1C&ref=sharing
https://www.facebook.com/reel/606490164289064?extid=a&s=yWDuG2&fs=e

### PR DESCRIPTION
**Pull request by bramaudi**
Closes #54

### CODE REVIEW

**Comment by JakeyPrime** [comment]
Tested in my install, works like a charm

**Comment by hasyim404** [comment]
works well before, but the link like **https://www.facebook.com/share/v/iz28EXJSeQbBQ4yG/** is not supported, 

image

if u copy link in application (not web/browser), the output have "share" in the url, and cause invalid url

**Comment by bramaudi** [comment]
@hasyim404 I have pushed a fix for your problem and have tested it successfully, but unfortunately with the URL you provided I got a different video with cobalt but not with browser, I'm not sure what causing this.

Please give it a try with different "shared link" they should work as expected.

File: src/modules/processing/url.js [hunk 1]
```diff
@@ -23,15 +23,15 @@ export function aliasURL(url) {
...
+                    }`)
             }
             break;
 
         case "pin":
             if (url.hostname === 'pin.it' && parts.length === 2) {
                 url = new URL(`https://pinterest.com/url_shortener/${
                     encodeURIComponent(parts[1])
-                }`)
+                    }`)
```
Comment (dumbmoron) [inline src/modules/processing/url.js:34]: why?

Comment (bramaudi) [reply src/modules/processing/url.js:34]: it seem I accidentally do auto-formating with my IDE

File: src/modules/processing/services/facebook.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    const urls = []
+    const hd = html.match('"browser_native_hd_url":"(.*?)"')
+    const sd = html.match('"browser_native_sd_url":"(.*?)"')
+
+    if (hd?.length) {
+        urls.push(JSON.parse(`["${hd[1]}"]`)[0])
+    }
+    if (sd?.length) {
+        urls.push(JSON.parse(`["${sd[1]}"]`)[0])
+    }
```
Comment (dumbmoron) [inline src/modules/processing/services/facebook.js:58]: this is unnecessarily complicated

```suggestion
    const hd = html.match('"browser_native_hd_url":(".*?")')
    const sd = html.match('"browser_native_sd_url":(".*?")')

    if (hd?.[1]) {
        urls.push(JSON.parse(hd[1]))
    }
    if (sd?.[1]) {
        urls.push(JSON.parse(sd[1]))
    }
```
also works fine

File: src/modules/processing/services/facebook.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    if (!urls.length) {
+        return { error: 'ErrorEmptyDownload' };
+    }
+
+    let filename = `facebook_${id}.mp4`
+    if (isShortLink) {
+        filename = `facebook_${shortLink}.mp4`
```
Comment (dumbmoron) [inline src/modules/processing/services/facebook.js:66]: i would just use the `id` for the filename in all cases

**Review by dumbmoron — DISMISSED** [review]

File: src/modules/processing/services/facebook.js [hunk 4]
```diff
@@ -26,19 +26,14 @@ function resolveUrl(url) {
...
 
-    if (isShortLink) {
-        url = await resolveUrl(url)
-    }
-
-    if (isSharedLink) {
-        url = sourceUrl
-    }
+    if (isShortLink) url = await resolveUrl(url)
+    if (isSharedLink) url = (sourceUrl)
```
Comment (dumbmoron) [inline src/modules/processing/services/facebook.js:36]:
```suggestion
    if (isSharedLink) url = sourceUrl
```

**Review by dumbmoron — APPROVED** [review]

**Pull request merged**